### DISCUSSION TITLE
Unable to update Vaultwarden

### DISCUSSION THREAD

**Discussion by benjaminbalo** [tteck/Proxmox#717] [Help Needed] [answered]
Hi,
first I want to say "Thank you" for this great and time-saving work.

Today i have to adress maybe an issue with the vaultwarden update script.
I see in the admin interface, that an update is available.
So, i went over to my cli and start the update-script with
`bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/misc/vaultwarden-update.sh)"`

Everything is fine and the script do its magic.

After finishing (and applying some os-related updates also) i rebooted the lxc; checking the vaultwarden-webui and the version is still the same old version. 

Installed: 1.25.2-60ed5ff9
Latest:  1.26.0-7a767310

Maybe you can have a check,if there is something wrong.

If you need any further informations, please let me know!

Thank you and have a nice day,

Ben

**Comment by tteck** [comment]
Just so happens that I also needed to update.

![Screenshot 2022-11-10 9 18 40 AM](https://user-images.githubusercontent.com/21342670/201119362-e78c0d53-a9a7-41af-a522-d56f861c7945.png)

I ran both update scripts and had no issues.

![Screenshot 2022-11-10 9 32 15 AM](https://user-images.githubusercontent.com/21342670/201119561-1528c70e-a332-429f-92c7-e5d43e03e62e.png)

Maybe try clearing your browser cache.

**Comment by benjaminbalo** [comment]
ok, this is strange....
no, i get this error on cli:
![grafik](https://user-images.githubusercontent.com/1911736/201294635-5f667756-75d5-4739-884b-32a17de6db23.png)

does your script poduce a log file?

**Reply by tteck** [reply]
Did you install Vaultwarden before 2022-05-29?

**Reply by benjaminbalo** [reply]
yes, i think so.
did the script change  between?

**Reply by tteck** [reply]
Yes, that's why it says **"**⚙️ To Update Vaultwarden (***post 2022-05-29 installs only***)**"**

**Comment by benjaminbalo** [comment]
oups....ok, means i need to create a new instance of it...sorry for bothering.
I'll go now looking for exporting the database.... thanks for helping :)

**Reply by tteck** [reply]
Just export personal vault and organization vault from your current Vaultwarden LXC
import personal vault and organization vault in your new Vaultwarden LXC

**Reply by benjaminbalo** [reply]
doesn't help my users... :( have to export the whole thing....

**Reply by tteck** [reply]
I elected to do it this way because a lot changed (a complete script re-write)
I can try and add the old valtwarden path to see if you can update, let me know

**Reply by tteck** [reply]
Try
```

```

**Reply by tteck** [reply]
If that works for you, I'll add the changes to the main branch so the next time you update, no problems

**Reply by benjaminbalo** [reply]
oh my gosh! it worked!
thank you soooo much! :D 
i really appreciate your help!

**Comment by chx0003** [comment]
Remember to set the RAM of the LXC to 3072Mb (instead of 2048Mb) before running the update script, otherwise it may not build correctly and the script won't find the file "target/release/vaultwarden" to copy to the program folder...

**Reply by benjaminbalo** [reply]
thanks for the hint :)

**Comment by tteck** [comment, marked as answer]
Add check for old path in main branch

**Comment by timof-de** [comment]
Hey,
question from my side: How long does the update process run?
Already stuck for about 2 hours at this step of the script:
```
This will Update Vaultwarden to 1.26.0 (set 2vCPU 3072MiB RAM Min.). Proceed(y/n)?y
 Stopping Vaultwarden... 
 Updating (Building) to 1.26.0 (Patience)... 
```
Memory is nearly at 100% of 512 MB
SWAP ist permament at 100% of 512 MB

Should I have manually expanded the memory before starting the script?

**Reply by tteck** [reply]
Well, the update script says "This will Update Vaultwarden to 1.26.0 (**set 2vCPU 3072MiB RAM Min**.). Proceed(y/n)?"

**Reply by timof-de** [reply]
Thanks for your quick response. My interpretation was that the script does this automatically as during the installation.
Then I dare to try again after breaking off the routine ...

**Reply by tteck** [reply]
you can try to set resources now while it's running to see if it'll finish. I'm surprised it hasn't given an error.

**Reply by tteck** [reply]
> My interpretation was that the script does this automatically

Can't set resources from the LXC console.

**Reply by tteck** [reply]
Made changes to clear confusion, "(Ensure you set 2vCPU 3072MiB RAM Min.)"

**Reply by timof-de** [reply]
Thanks, might be useful for the Proxmox-noobs like me. :)
Stopping the script, changing the setting and running the script again works for me. Takes about 10-15 min for the update.
